rebasehelper/checkers/abipkgdiff_tool.py

```diff
@@ -113,19 +116,90 @@ def run_check(cls, result_dir, **kwargs):
             command.append(pkg)
             command.append(new_pkg)
             logger.debug('Package name for ABI comparison %s', old_name)
-            output = os.path.join(cls.results_dir, result_dir, old_name + '-' + cls.log_name)
+            output = os.path.join(cls.results_dir, old_name + '.txt')
             try:
                 ret_code = ProcessHelper.run_subprocess(command, output=output)
             except OSError:
                 raise CheckerNotFoundError("Checker '%s' was not found or installed." % cls.CMD)
 
             if int(ret_code) & settings.ABIDIFF_ERROR and int(ret_code) & settings.ABIDIFF_USAGE_ERROR:
                 raise RebaseHelperError('Execution of %s failed.\nCommand line is: %s' % (cls.CMD, cmd))
-            if int(ret_code) == 0:
-                text = 'ABI of the compared binaries in package %s are equal.' % old_name
-            else:
-                text = 'ABI of the compared binaries in package %s are not equal.' % old_name
-                cls.critical_changes = True
-            reports[output] = text
-        reports['critical_changes'] = cls.critical_changes
-        return reports
+            ret_code = int(ret_code)
+            reports[old_name] = ret_code
+        return {'packages': cls.parse_abi_logs(reports), 'abi_changes': cls.abi_changes, 'path': cls.results_dir}
+
+    @classmethod
+    def parse_abi_logs(cls, reports):
+        """
+        Parse abipkgdiff logs
+        :param reports: dictionary with paths to the logs of the produced (sub)packages
+        :return: returns packages names with its abipkgdiff changes
+        """
+        pkgs = {}
+        for pkg, ret_code in six.iteritems(reports):
+            cur_pkg = {}
+            if ret_code:
+                with open(os.path.join(cls.results_dir, pkg + '.txt'), 'r') as f:
+                    lines = f.readlines()
+                    cur_pkg['functions_changes'] = cls.parse_line("Functions changes .* (\d+) .* (\d+) .* (\d+)",
+                                                                 lines[1])
+                    cur_pkg['variables_changes'] = cls.parse_line("Variables changes .* (\d+) .* (\d+) .* (\d+)",
+                                                                 lines[2])
+                    cur_pkg['functions_symbols_changes'] = cls.parse_line("Function symbols .* (\d+) .* (\d+)",
+                                                                         lines[3])
+                    cur_pkg['variables_symbols_changes'] = cls.parse_line("Variable symbols .* (\d+) .* (\d+)",
+                                                                          lines[4])
```

Comment:
The regexps should be definitely more specific. This doesn't have any better chance to match correctly if something changes IMHO. On the other hand, I don't see any reason for matching specific patterns on specific lines. If the order changes it could still work.

Also, you are using single backslashes in non-raw string literals, which is wrong.

Reply:
```
Functions changes summary: 0 Removed, 1 Changed (2 filtered out), 0 Added function
```
We could look for the `changes summary` string and display the whole line, if any changes were found.